### ISSUE TITLE
systemd-journald: `Program terminated with signal SIGABRT, Aborted.`

### ISSUE THREAD

**Issue by paulepanter** [systemd/systemd#2899]
### Submission type
- [x] Bug report
- [ ] Request for enhancement (RFE)

_NOTE: Do not submit anything other than bug reports or RFEs via the issue tracker!_
### systemd version the issue has been seen with

```
$ systemd --version
systemd 229
+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ -LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN
$ dpkg -l systemd
Gewünscht=Unbekannt/Installieren/R=Entfernen/P=Vollständig Löschen/Halten
| Status=Nicht/Installiert/Config/U=Entpackt/halb konFiguriert/
         Halb installiert/Trigger erWartet/Trigger anhängig
|/ Fehler?=(kein)/R=Neuinstallation notwendig (Status, Fehler: GROSS=schlecht)
||/ Name                          Version             Architektur         Beschreibung
+++-=============================-===================-===================-===============================================================
ii  systemd                       229-3               i386                system and service manager
$ uname -a
Linux myasrocke350m1 3.19.0-trunk-686-pae #1 SMP Debian 3.19-1~exp1 (2015-02-12) i686 GNU/Linux
```

_NOTE: Do not submit bug reports about anything but the two most recently released systemd versions upstream!_
### Used distribution

> Debian Sid/unstable
### In case of bug report: Expected behaviour you didn't see

> No crash.
### In case of bug report: Unexpected behaviour you saw

> The program `systemd-journald` crashed.

```
Mär 25 09:42:20 myasrocke350m1 systemd-journald[8197]: System journal (/var/log/journal/) is 1.0G, max 995.1M, 0B free.
Mär 25 09:42:23 myasrocke350m1 systemd[1]: systemd-journald.service: Main process exited, code=dumped, status=6/ABRT
Mär 25 09:42:23 myasrocke350m1 systemd[1]: systemd-journald.service: Unit entered failed state.
Mär 25 09:42:23 myasrocke350m1 systemd[1]: systemd-journald.service: Failed with result 'core-dump'.
Mär 25 09:42:23 myasrocke350m1 systemd[1]: systemd-journald.service: Service has no hold-off time, scheduling restart.
Mär 25 09:42:23 myasrocke350m1 systemd[1]: Stopped Flush Journal to Persistent Storage.
Mär 25 09:42:23 myasrocke350m1 systemd[1]: Stopping Flush Journal to Persistent Storage...
Mär 25 09:42:23 myasrocke350m1 systemd[1]: Stopped Journal Service.
Mär 25 09:42:23 myasrocke350m1 systemd[1]: Starting Journal Service...
Mär 25 09:42:23 myasrocke350m1 systemd-coredump[8193]: Detected coredump of the journal daemon itself, diverted to /var/lib/systemd/coredump/cor
Mär 25 09:42:23 myasrocke350m1 systemd-journald[8197]: Journal started
Mär 25 09:42:23 myasrocke350m1 systemd[1]: Started Journal Service.
Mär 25 09:39:01 myasrocke350m1 CRON[8026]: pam_unix(cron:session): session opened for user root by (uid=0)
Mär 25 09:39:01 myasrocke350m1 systemd[1]: Created slice system-systemd\x2dcoredump.slice.
Mär 25 09:39:01 myasrocke350m1 CRON[8027]: (root) CMD (  [ -x /usr/lib/php5/sessionclean ] && /usr/lib/php5/sessionclean)
Mär 25 09:39:01 myasrocke350m1 systemd[1]: Started Process Core Dump (PID 8024/UID 0).
Mär 25 09:39:03 myasrocke350m1 CRON[8026]: pam_unix(cron:session): session closed for user root
Mär 25 09:41:15 myasrocke350m1 systemd[1]: systemd-journald.service: Watchdog timeout (limit 3min)!
Mär 25 09:40:07 myasrocke350m1 smartd[814]: Device: /dev/sda [SAT], SMART Usage Attribute: 190 Airflow_Temperature_Cel changed from 84 to 74
Mär 25 09:42:23 myasrocke350m1 systemd[1]: Starting Flush Journal to Persistent Storage...
Mär 25 09:42:23 myasrocke350m1 systemd[1]: Started Flush Journal to Persistent Storage.
Mär 25 09:44:00 myasrocke350m1 wpa_supplicant[1231]: wlan0: WPA: Group rekeying completed with 34:31:c4:bd:c8:97 [GTK=CCMP]
Mär 25 09:44:01 myasrocke350m1 systemd[1]: systemd-coredump@0-8024-0.service: Service reached runtime time limit. Stopping.
Mär 25 09:44:01 myasrocke350m1 systemd[1]: systemd-coredump@0-8024-0.service: Unit entered failed state.
Mär 25 09:44:01 myasrocke350m1 systemd[1]: systemd-coredump@0-8024-0.service: Failed with result 'timeout'.
Mär 25 09:44:30 myasrocke350m1 gnome-session-binary[1291]: WARNING: Lost name on bus: org.gnome.SessionManager

```

```
$ gdb /lib/systemd/systemd-journald
GNU gdb (Debian 7.10-1+b1) 7.10
Copyright (C) 2015 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "i586-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /lib/systemd/systemd-journald...Reading symbols from /usr/lib/debug/.build-id/46/1e43fcde0f289d83b20938d66ef757b2ee84a4.debug...done.
done.
(gdb) set pagination off
(gdb) core-file coredumps/core.systemd-journal.0.985637b165a141f0812af1481b8d588c.329.1458895294000000
warning: core file may not match specified executable file.
[New LWP 329]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/i386-linux-gnu/i686/cmov/libthread_db.so.1".
Core was generated by `/lib/systemd/systemd-journald'.
Program terminated with signal SIGABRT, Aborted.
#0  0xb7729bcc in __kernel_vsyscall ()
(gdb) bt f
#0  0xb7729bcc in __kernel_vsyscall ()
No symbol table info available.
#1  0xb75ac3dc in __fsync_nocancel () at ../sysdeps/unix/syscall-template.S:84
No locals.
#2  0xb7766689 in journal_file_set_offline (f=0xb86aac78) at ../src/journal/journal-file.c:127
        __PRETTY_FUNCTION__ = "journal_file_set_offline"
#3  0xb775c20c in server_sync (s=0xbfc318a0) at ../src/journal/journald-server.c:375
        f = 0xb86aa218
        i = {idx = 3078241492, next_key = 0xb86a919c}
        r = <optimized out>
        __func__ = "server_sync"
#4  0xb7785b1c in server_dispatch_sync (es=0xb86b2a60, t=1813998424, userdata=0xbfc318a0) at ../src/journal/journald-server.c:1460
        s = 0xbfc318a0
        __PRETTY_FUNCTION__ = "server_dispatch_sync"
#5  0xb777f940 in source_dispatch.lto_priv.278 (s=0xb86b2a60) at ../src/libsystemd/sd-event/sd-event.c:2273
        r = <optimized out>
        __PRETTY_FUNCTION__ = "source_dispatch"
        __func__ = "source_dispatch"
#6  0xb77556d5 in sd_event_dispatch (e=0xb86a9150) at ../src/libsystemd/sd-event/sd-event.c:2625
        p = <optimized out>
        r = <optimized out>
#7  sd_event_run (timeout=<optimized out>, e=0xb86a9150) at ../src/libsystemd/sd-event/sd-event.c:2684
        r = <optimized out>
#8  main (argc=1, argv=0xbfc31e44) at ../src/journal/journald.c:101
        t = <optimized out>
        n = <optimized out>
        server = {syslog_fd = 5, native_fd = 4, stdout_fd = 3, dev_kmsg_fd = 9, audit_fd = 6, hostname_fd = 10, notify_fd = 14, event = 0xb86a9150, syslog_event_source = 0xb86a9440, native_event_source = 0xb86a9510, stdout_event_source = 0xb86a9370, dev_kmsg_event_source = 0xb86a95e0, audit_event_source = 0xb86a96b0, sync_event_source = 0xb86b2a60, sigusr1_event_source = 0xb86a9958, sigusr2_event_source = 0xb86a9a28, sigterm_event_source = 0xb86a9af8, sigint_event_source = 0xb86a9c70, sigrtmin1_event_source = 0xb86a9d50, hostname_event_source = 0xb86a9780, notify_event_source = 0xb86aa130, watchdog_event_source = 0xb86aa270, runtime_journal = 0x0, system_journal = 0xb86aac78, user_journals = 0xb86a8154, seqnum = 285590, buffer = 0xb86aafc8 "<86>Mar 25 09:34:37 sudo: pam_unix(sudo:session): session closed for user root", buffer_size = 24576, rate_limit = 0xb86a9ed8, sync_interval_usec = 300000000, rate_limit_interval = 30000000, rate_limit_burst = 1000, runtime_metrics = {max_size = 20189184, min_size = 524288, max_use = 161509376, min_use = 8388608, keep_free = 242262016, n_max_files = 100}, system_metrics = {max_size = 130437120, min_size = 524288, max_use = 1043472384, min_use = 1043472384, keep_free = 1565208576, n_max_files = 100}, compress = true, seal = true, forward_to_kmsg = false, forward_to_syslog = true, forward_to_console = false, forward_to_wall = true, n_forward_syslog_missed = 0, last_warn_forward_syslog_missed = 0, cached_space_available = 0, cached_space_limit = 1043472384, cached_space_timestamp = 1796576792, var_available_timestamp = 0, max_retention_usec = 0, max_file_usec = 2629800000000, oldest_file_usec = 1442554026865348, stdout_streams = 0xb86d4f00, stdout_streams_notify_queue = 0x0, n_stdout_streams = 61, tty_path = 0x0, max_level_store = 7, max_level_syslog = 7, max_level_kmsg = 5, max_level_console = 6, max_level_wall = 0, storage = STORAGE_AUTO, split_mode = SPLIT_UID, mmap = 0xb86a7028, udev = 0xb86a9ec8, kernel_seqnum = 0xb7724000, dev_kmsg_readable = true, send_watchdog = false, sent_notify_ready = true, sync_scheduled = true, machine_id_field = "_MACHINE_ID=1572dc2ca76ca04c3351599547f539a6", boot_id_field = "_BOOT_ID=985637b165a141f0812af1481b8d588c", hostname_field = 0xb86aa110 "_HOSTNAME=mattotaupa", cgroup_root = 0xb86aa260 "", watchdog_usec = 180000000}
        r = <optimized out>
        __func__ = "main"
```
### In case of bug report: Steps to reproduce the problem

> No idea. Just let a lot of programs run in parallel to cause a lot of load.

**Comment by paulepanter** [comment]
I guess this is a duplicate of issue #917, which I submitted last year.

My problem is still, that in this case, GDM restarts and my whole session is gone. Also, even if it’s a Linux kernel problem, what do people with older Linux kernel without the fix do?

**Comment by paulepanter** [comment]
Sorry, and I forget to mention, that this time there are no Linux messages in the journal talking about a stall or block.

**Comment by poettering** [comment]
Can you reproduce this reliably? Do you see any issue after this happens mentioned in dmesg?

If you can reproduce this, could you first use "systemd-analyze set-log-level debug", and then reproduce it? This should give more information in the logs from PID 1 itself what's going on.

**Comment by poettering** [comment]
Closing due to lack of response. Please reopen if this is still an issue and you can provide the requested data! Thanks!
